### PR TITLE
Fixes chat and traps not shown

### LINKED ISSUE #55: Missing model for the cannon trap
The cannon trap can be placed in-game, but it has no model, so it's a bit confusing. See:

odscreenshot_22
The claimed tile under the cursor (in the middle of the training area) is my cannon trap, and it just fired a cannonball at a drake! :)

### PR DESCRIPTION
**Pull request by hwoarangmy**
fixes #55

### CODE REVIEW

File: source/ODPacket.h [hunk 1]
```diff
@@ -29,6 +29,39 @@
  * ODPacket should preserve integrity. That means that if an ODSocketClient
  * sends an ODPacket, the server should receive exactly 1 similar ODPacket (same data,
  * nothing less, nothing more)
+ * IMPORTANT : ODPacket can handle different types of data the same way. For that, when we receive
```
Comment (Bertram25) [inline source/ODPacket.h:32]: Just thanks for this. :)
I can swear It will avoid some later mess.

File: source/ODServer.h [hunk 2]
```diff
@@ -25,6 +25,24 @@
 class ServerNotification;
 class GameMap;
 
+/**
```
Comment (Bertram25) [inline source/ODServer.h:28]: Thanks also for the explanation about the client/server paradigm.
We'll be able to point to this whenever someone doesn't seem to get how this works.

**Comment by Bertram25** [comment]
Thanks also for sorting out the messy handling of OIS keyboard translation. The code is looking good and even streamlined now. :)
I'll test this asap on linux and merge if everything is ok.

**Comment by hwoarangmy** [comment]
Concerning the messy handling of OIS keyboard translation, there might be some regression somewhere (probably in the console since translation removing was there). I guess the changes were not done for nothing even if I could find no reason why. I've asked on the forum if someone knows why but it is still broken so I don't know if someone answered. Anyway, I just though that because chat is a game feature, we should do what is needed to make it work. If we need to do some hacks, I think it is better to do them in the console rather than in a game feature.

**Comment by Bertram25** [comment]
> If we need to do some hacks, I think it is better to do them in the console rather than in a game feature.

Agreed.

> I've asked on the forum if someone knows why but it is still broken so I don't know if someone answered.

I'm relieved I'm not the only one failing at reaching the forums. :)

**Comment by akien-mga** [comment]
> I'm relieved I'm not the only one failing at reaching the forums. :)

FYI: http://freegamer.blogspot.fr/2014/08/freegamedevnet-outage.html

**Comment by Bertram25** [comment]
@akien-mageia I won't be able to test this before this evening. I wouldn't like to be too demanding, but if you can reach some linux computer before me, could you test this one in game?
If you can't, it's not a problem. :)

How-to: create a local branch, pull hwoarangmy's patch, compile, test in game that #48, #49, #50 and #55 are fixed. Check for regression on the console, for instance. (See the `command line` link below.)

```
git checkout -b hwoarangmy-development development
git pull git://github.com/hwoarangmy/OpenDungeons.git development
```

The code looks fine so once I'll have your ok, I'll be able to merge.

**Comment by akien-mga** [comment]
I can have a look at 6 p.m. Paris time, I'll do some thorough testing and breaking of the chat box at that time :-)

**Comment by Bertram25** [comment]
Thanks! :) It will be 3 hours before me. ;)

**Comment by akien-mga** [comment]
I get a linking issue at the end of the compilation (gcc 4.9.2 dev):

```
Linking CXX executable OpenDungeons.bin
CMakeFiles/OpenDungeons.bin.dir/source/MenuModeMultiplayer.cpp.o: In function `MenuModeMultiplayer::serverButtonPressed()':
MenuModeMultiplayer.cpp:(.text+0x891): undefined reference to `std::basic_string<char, std::char_traits<char>, std::allocator<char> > boost::locale::conv::to_utf<char>(char const*, char const*, std::string const&, boost::locale::conv::method_type)'
CMakeFiles/OpenDungeons.bin.dir/source/ODFrameListener.cpp.o: In function `ODFrameListener::notifyChatChar(int)':
ODFrameListener.cpp:(.text+0xe73): undefined reference to `std::basic_string<char, std::char_traits<char>, std::allocator<char> > boost::locale::conv::to_utf<char>(char const*, char const*, std::string const&, boost::locale::conv::method_type)'
collect2: error: ld returned 1 exit status
CMakeFiles/OpenDungeons.bin.dir/build.make:3573: recipe for target 'OpenDungeons.bin' failed
make[2]: *** [OpenDungeons.bin] Error 1
CMakeFiles/Makefile2:60: recipe for target 'CMakeFiles/OpenDungeons.bin.dir/all' failed
make[1]: *** [CMakeFiles/OpenDungeons.bin.dir/all] Error 2
Makefile:137: recipe for target 'all' failed
make: *** [all] Error 2
```

**Comment by akien-mga** [comment]
The locale component from boost needs to be explicitly linked, and it's missing for Unix:

```
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6a48259..b872471 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -285,5 +285,5 @@ find_package(OpenAL REQUIRED)
 if(UNIX)
     find_package(Threads REQUIRED)
-    find_package(Boost COMPONENTS thread system filesystem REQUIRED)
+    find_package(Boost COMPONENTS thread system filesystem locale REQUIRED)
 endif()
```

**Comment by akien-mga** [comment]
@Bertram25 You can merge the pull request once the CMakeLists.txt change I described above has been added to the pull request. All said bugfixes are indeed fixed :-)

**Comment by hwoarangmy** [comment]
I had not seen boost components were different between linux and windows. Strange because I also added filesystem and did not the same mistake. Anyway, I'm gonna make the change right now

**Comment by hwoarangmy** [comment]
ok, it's done. Thanks for the review akien

**Comment by Bertram25** [comment]
Note to self: Add the boost locale to the needed libraries.

**Comment by Bertram25** [comment]
As per review from Akien, I'm gladly merging. :) Nice work guys. I feel useless here, and it's somehow... great. :)

**Pull request merged**